### ISSUE TITLE
MPS torch.var cannot handle zero-dimensional input tensor

### ISSUE THREAD

**Issue by manuelcandales** [pytorch/pytorch#160738]
### 🐛 Describe the bug

As a general rule, torch operations allow accessing dimension 0 on a zero-dimensional tensor (returning back the same tensor). In this case, the CPU torch.var operation can handle zero-dimensional input with dim=0, but MPS throws error:

```python
import torch

def test_var(device):
    x = torch.tensor(3.0, device=device)
    try:
        output = torch.var(x, dim=0)
        print(f"var test succeeds for device: {device}. output: {output}")
    except Exception as e:
        print(f"var test fails for device: {device}: {e}")

test_var(device = "cpu")
test_var(device = "mps")
```

terminal output:
```
var test succeeds for device: cpu. output: nan
var test fails for device: mps: var_mps: reduction dim must be in the range of input shape
```

### Versions

nightly
